packages/amplify-provider-awscloudformation/src/push-resources.js

```diff
@@ -504,6 +639,18 @@ function formNestedStack(context, projectDetails, categoryName, resourceName, se
   const nestedStack = context.amplify.readJsonFile(initTemplateFilePath);
   const { amplifyMeta } = projectDetails;
   let authResourceName;
+
+  const { NetworkStackS3Url } = amplifyMeta.providers[constants.ProviderName];
+
+  if (NetworkStackS3Url) {
```

Comment:
Why do we need a separate logic for handling nested stacks for this resource specifically? The nested stack generation is currently handled in one place based of the amplify-meta.json. This would be a deviation from what we're currently doing.

Reply:
Nested stack deployed only once for customers and on subsequent projects that stack is only referenced.